crates/cairo-types/src/types/data_type.rs

```diff
@@ -314,6 +286,81 @@ mod tests {
         assert_eq!(u128::deserialize(&felts, 1).unwrap(), 99);
     }
 
+    #[test]
+    fn test_serialize_i8() {
+        let v = 123_i8;
+        let felts = i8::serialize(&v);
+        assert_eq!(felts.len(), 1);
+        assert_eq!(felts[0], FieldElement::from(123 as u8));
+    }
+
+    #[test]
+    fn test_deserialize_i8() {
+        let felts = vec![FieldElement::from(123 as u8), FieldElement::from(123 as u8)];
+        assert_eq!(i8::deserialize(&felts, 0).unwrap(), 123);
+        assert_eq!(i8::deserialize(&felts, 1).unwrap(), 123);
+    }
+
+    #[test]
+    fn test_serialize_i16() {
+        let v = 123_i16;
+        let felts = i16::serialize(&v);
+        assert_eq!(felts.len(), 1);
+        assert_eq!(felts[0], FieldElement::from(123 as u8));
+    }
+
+    #[test]
+    fn test_deserialize_i16() {
+        let felts = vec![FieldElement::from(123 as u8), FieldElement::from(123 as u8)];
+        assert_eq!(i16::deserialize(&felts, 0).unwrap(), 123);
+        assert_eq!(i16::deserialize(&felts, 1).unwrap(), 123);
+    }
+
+    #[test]
+    fn test_serialize_i32() {
+        let v = 123_i32;
+        let felts = i32::serialize(&v);
+        assert_eq!(felts.len(), 1);
+        assert_eq!(felts[0], FieldElement::from(123 as u8));
+    }
+
+    #[test]
+    fn test_deserialize_i32() {
+        let felts = vec![FieldElement::from(123 as u8), FieldElement::from(123 as u8)];
+        assert_eq!(i32::deserialize(&felts, 0).unwrap(), 123);
+        assert_eq!(i32::deserialize(&felts, 1).unwrap(), 123);
+    }
+
+    #[test]
+    fn test_serialize_i64() {
+        let v = 123_i64;
+        let felts = i64::serialize(&v);
+        assert_eq!(felts.len(), 1);
+        assert_eq!(felts[0], FieldElement::from(123 as u8));
+    }
+
+    #[test]
+    fn test_deserialize_i64() {
+        let felts = vec![FieldElement::from(123 as u8), FieldElement::from(123 as u8)];
+        assert_eq!(i64::deserialize(&felts, 0).unwrap(), 123);
+        assert_eq!(i64::deserialize(&felts, 1).unwrap(), 123);
+    }
+
+    #[test]
+    fn test_serialize_i128() {
+        let v = 123_i128;
+        let felts = i128::serialize(&v);
+        assert_eq!(felts.len(), 1);
+        assert_eq!(felts[0], FieldElement::from(123 as u8));
+    }
+
+    #[test]
+    fn test_deserialize_i128() {
+        let felts = vec![FieldElement::from(123 as u8), FieldElement::from(123 as u8)];
+        assert_eq!(i128::deserialize(&felts, 0).unwrap(), 123);
+        assert_eq!(i128::deserialize(&felts, 1).unwrap(), 123);
+    }
+
```

Comment:
Could you please change the tests to use `i8::MAX, i16::MAX` etc..? Because for now tests pass, because only 123 is used. :)

Also, we may consider a macro to also test the ints/uints, using the `$type::MAX` as the value to be tested. :+1: